### PR TITLE
Revisit laws and fix Optional.first and second

### PR DESCRIPTION
**Pull request by julien-truffaut**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -56,6 +56,8 @@ lazy val macroCompat = Def.setting("org.typelevel" %%% "macro-compat" % "1.1.0")
 lazy val macroVersion = "2.1.0"
 lazy val paradisePlugin = compilerPlugin("org.scalamacros" %  "paradise"       % macroVersion cross CrossVersion.full)
 
+lazy val kindProjector = "org.spire-math" % "kind-projector" % "0.7.1" cross CrossVersion.binary
```
Comment (aoiroaoino) [inline build.sbt]: coooool 👍

**Comment by julien-truffaut** [comment]
@aoiroaoino @NightRa when you get the time, would you mind to have a look at this PR.

I am working on upgrading scalacheck such as we can benefit from the new `CoGen` and so have good random function. However, I would prefer to do that in a separate PR.

File: core/shared/src/main/scala/monocle/law/LensLaws.scala [hunk 2]
```diff
@@ -14,12 +15,18 @@ class LensLaws[S, A](lens: Lens[S, A]) {
   def setGet(s: S, a: A): IsEq[A] =
     lens.get(lens.set(a)(s)) <==> a
 
-  def setIdempotent(s: S, a: A): IsEq[S] =
-    lens.set(a)(lens.set(a)(s)) <==> lens.set(a)(s)
```
Comment (aoiroaoino) [inline core/shared/src/main/scala/monocle/law/LensLaws.scala]: Why remove it?

**Comment by aoiroaoino** [comment]
Sorry for the late reply. 😅 
I have an above question. other than that looks good. 👍

File: core/shared/src/main/scala/monocle/law/LensLaws.scala (hunk 2 above)
Comment (julien-truffaut) [inline core/shared/src/main/scala/monocle/law/LensLaws.scala]: I believe it is redundant with `modifyId` and `consistentSetModify` but I might be wrong

Comment (julien-truffaut) [inline core/shared/src/main/scala/monocle/law/LensLaws.scala]: In fact, I think I am wrong. I don't think `setIdempotent` is redundant. I will add it back

**Comment by aoiroaoino** [comment]
👍 👍 👍

**Pull request merged**